Client/game_sa/CSettingsSA.cpp

```diff
@@ -327,7 +328,22 @@ void CSettingsSA::SetVolumetricShadowsSuspended(bool bSuspended)
     m_bVolumetricShadowsSuspended = bSuspended;
 }
 
-//
+
+bool CSettingsSA::IsLightsAndShadowsImprovementEnabled()
+{
+    return m_bLightsAndShadowsImprovement;
+}
+
+void CSettingsSA::SetLightsAndShadowsImprovementEnabled(bool bEnable)
+{
+    m_bLightsAndShadowsImprovement = bEnable;
+
+    if (bEnable == true) 
+        MemPut<DWORD>(0x00859934, 1018127695);
+    else
+        MemPut<DWORD>(0x00859934, 1031127695);
```

Comment:
```cpp
m_bLightsAndShadowsImprovement = bEnable;
MemPut<DWORD>(0x00859934, bEnable ? 1018127695 : 1031127695);
```
Also I see in IDA this is a float value. Im pretty sure you can just use `MemPut<float>`, or?